draft-ietf-add-split-horizon-authority.xml

```diff
@@ -179,158 +185,163 @@
-      </section>
+    <section>
+      <name>Requirements</name>
+      <t>This solution seeks to fulfill the following requirements:</t>
+      <ul>
+        <li>No loss of security: No unauthorized party can impersonate
+          a zone unless they could already do so without use of this
+          specification.</li>
+        <li>Least privilege: Local resolvers do not hold any
+          secrets that could weaken the security of the public zone if
+          compromised.</li>
+        <li>Local zone confidentiality: The specification does not leak
+          local network subdomains to anyone outside of the network.</li>
+        <li>Flexibility: The specification can represent and authorize
+          any reasonable Split DNS zone structure.</li>
+        <li>DNSSEC Compatibility: The specification supports DNSSEC-based
+          object security for local zone contents.</li>
+      </ul>
     </section>
-    <!-- Learning -->
-
     <section anchor="establishing">
       <name>Establishing Local DNS Authority</name>
       <t>To establish its authority over some DNS zone, a
       participating network <bcp14>MUST</bcp14> offer one or more
       encrypted resolvers via DNR <xref target="I-D.ietf-add-dnr"/>,
       DDR <xref target="I-D.ietf-add-dnr"/>, or an equivalent mechanism
-      (see <xref target="vpn"/>).  If the encrypted resolver is
-      identified by name (e.g., DNR), at least one of these resolvers'
-      Authentication Domain Names (ADNs) <bcp14>MUST</bcp14> appear in
-      an NS record for that zone.  If the encrypted resolver is
-      identified by IP address (e.g., DDR), then the resolver MUST present a
-      certificate that can be validated by the client, containing at least
-      one subjectAltName entry that matches an NS record for that zone.
-      This arrangement establishes this resolver's authority over the zone.</t>
+      (see <xref target="vpn"/>).</t>
+      <t>To establish local authority, the network MUST convey one or more
+      "Authorization Claims" to the client.  An "Authorization Claim" is an
+      abstract structure comprising:</t>
+      <ul>
+        <li>An Authentication Domain Name (ADN) of a local encrypted resolver.</li>
+        <li>The DNS name of the authorizing parent zone.</li>
+        <li>A list of subdomains of this parent zone that are claimed by
+            the named local resolver (potentially including the entire parent
+            zone).</li>
+        <li>An NSEC3 Hash Algorithm.</li>
+        <li>A high-entropy salt.</li>
+      </ul>
+      <t>If the local encrypted resolver is identified by name (e.g., DNR), that
+      identifying name MUST be the one used in any corresponding Authorization
+      Claim.  Otherwise (e.g., DDR using IP addresses), the resolver MUST
+      present a validatable certificate containing a subjectAltName that
+      matches the Authorization Claim.</t>
+      <t>To establish its authority, the network MUST provide each Authorization
```

Comment:
We may want to consider adding some text to say the draft requires the DNS administrators running the public zone and local DNS to co-operate for establishing local DNS authority in split-horizon environments.

Reply:
I see the comment raised in the WG that DNS people running the public zone and the split-DNS people do not typically communicate with each other. We can consider adding the following text to Introduction:
This document requires the DNS administrators running the public zone and local zone to co-operate for establishing local DNS authority in split-horizon environments.

Reply:
do you mean "publicly visible zone view" ?

Reply:
> If adding that text would help get us to WG consensus, I'm happy to add it.

Let's add the text to illustrate the expected modus operandi.

Reply:
Done